oak/server/rust/oak_runtime/src/node/grpc/client.rs

```diff
@@ -180,38 +245,36 @@ impl GrpcClientNode {
                 // Send the response back to the invocation channel.
                 debug!("Sending gRPC response: {:?}", encap_rsp);
                 invocation
-                    .send_response(encap_rsp, &runtime)
+                    .send_response(encap_rsp, runtime)
                     .map_err(|error| {
                         error!("Couldn't send gRPC response to the invocation: {:?}", error);
                         error
                     })?;
+                runtime
+                    .metrics_data()
+                    .grpc_client_metrics
+                    .observe_new_response_message(&self.method_name, msg_len);
             } else {
                 debug!("No message available, close out method invocation");
                 break;
             }
         }
-
         Ok(())
     }
+}
 
-    /// Creates a TLS connection to an external gRPC service.
-    async fn connect(
-        &self,
-    ) -> Result<tonic::client::Grpc<tonic::transport::channel::Channel>, tonic::transport::Error>
-    {
-        debug!("Connecting to {}", self.uri);
-
-        // Create a TLS configuration.
-        let tls_config = ClientTlsConfig::new().ca_certificate(self.root_tls_certificate.clone());
-
-        // Connect to a remote gRPC service.
-        let connection = Channel::builder(self.uri.clone())
-            .tls_config(tls_config)
-            .connect()
-            .await?;
-
-        debug!("Connected to {}", self.uri);
-        Ok(tonic::client::Grpc::new(connection))
+/// Manual implementation of the `Drop` trait to correctly update the metrics after the completion
+/// of the request handling.
```

Comment:
nit: "of the request handling" is slightly confusing, given that this is a type called `ResponseHandler`